benchmark/Cpp/Savina/src/parallelism/RadixSort.lf

```diff
@@ -0,0 +1,197 @@
+        }
+    =}
+}
+
+
+reactor SortActor(numValues:int(100000), bank_index:int(0)) {
+    
+    public preamble {=
+        #include <list>
+        #include "reactor-cpp/logging.hh"
+    =}
+    
+    state orderingArray:{=std::vector<reactor::ImmutableValuePtr<long>>=};
+    state valuesSoFar:int(0);
+    state radix:{=long=}(-1);
+    state pos:int(0);
+    state sendPos:int(0);
+    
+    input inValue:{=long=};
+    output outValue:{=long=};
+    
+    logical action sendBufferedValues:void;
+    
+    reaction(startup) {=
+        radix = 1l << bank_index;
+        orderingArray = std::vector<reactor::ImmutableValuePtr<long>>(numValues);
+    =}
+    
+    reaction(inValue) -> outValue, sendBufferedValues {=
+        
+        valuesSoFar += 1;
+        long current = *(inValue.get());
+        if((current & radix) == 0) {
+            // number with leading 0 (small) go through
+            outValue.set(current);
+        } else {
+            orderingArray[pos] = (inValue.get());
+            pos += 1;
+        }
+        
+        if(valuesSoFar == numValues) {
+            sendBufferedValues.schedule();
+        }
+    =}
+    
+    reaction(sendBufferedValues) -> outValue, sendBufferedValues {=
+        
+        if(sendPos < pos) {
+            outValue.set(orderingArray[sendPos]);
+            sendPos += 1;
+            sendBufferedValues.schedule();
+        } else {
+            
+            // reset local state
+            valuesSoFar = 0;
+            pos = 0;
+            sendPos = 0;
+        }
+    =}
+}
+
+reactor ValidationReactor(numValues:int(100000)) {
+    
+    private preamble {=
+        #include "reactor-cpp/logging.hh"
+    =}
+    
+    state sumSoFar:double(0.0);
+    state valuesSoFar:int(0);
+    state prevValue:{=long=}(0);
+    state errorValueLong:{=long=}(-1);
+    state errorValueInt:int(-1);
+    
+    input inValue:{=long=};
+    output outFinished:void;
+    
+    reaction(inValue) -> outFinished {=
+        
+        long value = *inValue.get();
+        valuesSoFar += 1;
+        if(value < prevValue && errorValueLong < 0) {
+            errorValueLong = value;
+            errorValueInt = valuesSoFar - 1;
+        }
+        prevValue = value;
+        sumSoFar += prevValue;
+        
+        if(valuesSoFar == numValues) {
+            if(errorValueLong >= 0) {
+                reactor::log::Error() << "ERROR: Value out of place: " << errorValueLong << " at index " << errorValueInt;
+            } else {
+                reactor::log::Info() << "Elements sum: " << sumSoFar;
+            }
+            
+            // reset local state
+            sumSoFar = 0.0;
+            valuesSoFar = 0;
+            prevValue = 0;
+            errorValueLong = -1;
+            errorValueInt = -1;
+            
+            outFinished.set();
+        }
+    =}
+}
+
+// FIXME We cannot adjust maxValue here because the number of sort actors below depends on it. The parameter should
```

Comment:
Implementing this change doesn't strike me as objectionable. How we deal with parameters isn't really a material part of the algorithm/workload.

Reply:
The problem here is rather the uniformity of parameter sets that the benchmark runner assumes. We could fix it here, but then the Savina benchmark would expect a different set of parameters. Dealing with this is currently not easily doable with our benchmark runner. It might become possible with a few changes though.